src/Storages/PostgreSQL/PostgreSQLReplicaConsumer.cpp

```diff
@@ -0,0 +1,682 @@
+            {
+                bool read_next = true;
+                switch (identifier)
+                {
+                    case 'K':
+                    {
+                        /// Only if changed column(s) are part of replica identity index (for now it can be only
+                        /// be primary key - default values for replica identity index). In this case, first comes a tuple
+                        /// with old replica identity indexes and all other values will come as nulls. Then comes a full new row.
+                        readTupleData(buffer->second, replication_message, pos, size, PostgreSQLQuery::UPDATE, true);
+                        break;
+                    }
+                    case 'O':
+                    {
+                        /// Old row. Only if replica identity is set to full. (For the case when a table does not have any
+                        /// primary key, for now not supported, requires to find suitable order by key(s) for nested table.)
+                        readTupleData(buffer->second, replication_message, pos, size, PostgreSQLQuery::UPDATE, true);
+                        break;
+                    }
+                    case 'N':
+                    {
+                        readTupleData(buffer->second, replication_message, pos, size, PostgreSQLQuery::UPDATE);
+                        read_next = false;
+                        break;
+                    }
+                }
+
+                return read_next;
+            };
+
+            /// Read either 'K' or 'O'. Never both of them. Also possible not to get both of them.
+            bool read_next = proccess_identifier(readInt8(replication_message, pos, size));
+
+            /// 'N'. Always present, but could come in place of 'K' and 'O'.
+            if (read_next)
+                proccess_identifier(readInt8(replication_message, pos, size));
+
+            break;
+        }
+        case 'D': // Delete
+        {
+            Int32 relation_id = readInt32(replication_message, pos, size);
+
+            if (!isSyncAllowed(relation_id))
+                return;
+
+             /// 0 or 1 if replica identity is set to full. For now only default replica identity is supported (with primary keys).
+            readInt8(replication_message, pos, size);
+
+            const auto & table_name = relation_id_to_name[relation_id];
+            auto buffer = buffers.find(table_name);
+            readTupleData(buffer->second, replication_message, pos, size, PostgreSQLQuery::DELETE);
+
+            break;
+        }
+        case 'C': // Commit
+        {
+            readInt8(replication_message, pos, size);  /// unused flags
+            readInt64(replication_message, pos, size); /// Int64 commit lsn
+            readInt64(replication_message, pos, size); /// Int64 transaction end lsn
+            readInt64(replication_message, pos, size); /// Int64 transaction commit timestamp
```

Comment:
Returned values are not used, so we can replace it with something like that:
```suggestion
            constexpr size_t unused_flags_len = 1;
            constexpr size_t commit_lsn_len = 8;
            constexpr size_t transaction_end_lsn_len = 8;
            constexpr size_t transaction_commit_timestamp_len = 8;
            pos += unused_flags_len + commit_lsn_len + transaction_end_lsn_len + transaction_commit_timestamp_len;
```

Reply:
Fixed.